ui/components/app/collectibles-items/collectibles-items.js

```diff
@@ -39,22 +43,44 @@ export default function CollectiblesItems({
   const collectionsKeys = Object.keys(collections);
   const collectiblesDropdownState = useSelector(getCollectiblesDropdownState);
   const previousCollectionKeys = usePrevious(collectionsKeys);
+  const selectedAddress = useSelector(getSelectedAddress);
+  const chainId = useSelector(getCurrentChainId);
 
   useEffect(() => {
     if (
-      !Object.keys(collectiblesDropdownState).length &&
-      previousCollectionKeys !== collectionsKeys
+      selectedAddress !== undefined &&
+      collectiblesDropdownState?.[selectedAddress] === undefined
+    ) {
+      collectiblesDropdownState[selectedAddress] = {};
```

Comment:
What does this empty object represent?

Reply:
We reach this block if collectibleDropDown state has not been initialized for the currently active address. This just initializes that state so that we can then safely initialize the state a layer deeper for the chainId as well. Because this state can be different across accounts and chains this state structure is nested two layers: selectedAddress -> chainId. This is also the shape for how we store state for collectibles data across accounts and chains. An example of what this looks like:

```{
    "0x806627172af48bd5b0765d3449a7def80d6576ff": {
        "0x1": {
            "0x495f947276749Ce646f68AC8c248420045cb7b5e": false
        },
        "0x4": {
            "0xf5de760f2e916647fd766B4AD9E85ff943cE3A2b": true
        }
    },
    "0xe7d522230eff653bb0a9b4385f0be0815420dd98": {
        "0x4": {
            "0xf5de760f2e916647fd766B4AD9E85ff943cE3A2b": false
        }
    }
}
```

Reply:
All that being said, you're right that we don't really need to initialize it like that. I've updated the logic to do this more cleanly: https://github.com/MetaMask/metamask-extension/pull/13435/commits/cbcbac0b785285b21ad2f79b7ea36bddeca433ee